src/components/VmDetail/index.js

```diff
@@ -156,10 +156,15 @@ class VmDetail extends Component {
       options,
       pool,
       onRDP,
+      operatingSystems,
     } = this.props
 
     const name = isPool ? pool.get('name') : vm.get('name')
-    const iconId = vm.getIn(['icons', 'small', 'id'])
+    let iconId = vm.getIn(['icons', 'small', 'id'])
+    const vmOs = operatingSystems.get('operatingSystems').find((v, k) => v.get('name') === vm.getIn(['os', 'type']))
+    if (vmOs) {
+      iconId = vmOs.getIn(['icons', 'large', 'id'])
```

Comment:
What if particular icon is not loaded?
Only the required icons are pre-fetched (means those already associated to VMs).
I suggest to keep logic as it is plus dispatch action to load icon of given ids (both small and large).

The saga performing such an action will do HTTP GET if particular iconId is missing or do nothing if already present.
Subsequent update of state enforces re-rendering.